### PR TITLE
[project-test] Fix issues from latest updates

### PR DESCRIPTION
**Pull request by jaylees14**
Some changes needed to make `temple test` work:

- Use `BASEDIR` everywhere
- Kill background processes before the end of the script (so it actually returns in Scala!)
- Fix name of kong script
- Fix formatting of dates/times etc.

Have tested with simple.temple - takes a very long time to spin up. Investigating using docker-compose for future modfications

### CODE REVIEW

**Review by Lewiky — APPROVED** [review]
`$BASEDIR/LGTM`

File: src/e2e/resources/simple-temple-expected/deploy.sh [hunk 1]
```diff
@@ -53,10 +53,13 @@ do
 done
 
 kubectl port-forward --namespace kube-system $REGISTRY_NAME 5000:5000 2>&1 1>/dev/null &
+PORT_FORWARD_PID=$!
 sleep 5
 
 # Push images to private repo
-sh push-image.sh
+sh $BASEDIR/push-image.sh
```
Comment (c-harding) [inline src/e2e/resources/simple-temple-expected/deploy.sh]: Quotes please, this doesn’t work in bash if BASEDIR has a space in it

File: src/e2e/resources/simple-temple-expected/deploy.sh [hunk 2]
```diff
@@ -39,7 +39,7 @@ do
 done
 
 # Create private registry
-kubectl create -f kube/deploy
+kubectl create -f $BASEDIR/kube/deploy
```
Comment (c-harding) [inline src/e2e/resources/simple-temple-expected/deploy.sh]: Ditto quotes

File: src/e2e/resources/simple-temple-expected/push-image.sh [hunk 3]
```diff
@@ -1,7 +1,8 @@
 #!/bin/sh
 REGISTRY_URL="localhost:5000"
+BASEDIR=$(dirname "$BASH_SOURCE")
 
 for service in "simple-temple-test-user" "booking" "simple-temple-test-group" "auth"; do
-  docker build -t "$REGISTRY_URL/simple-temple-test-$service" $service
+  docker build -t "$REGISTRY_URL/simple-temple-test-$service" $BASEDIR/$service
```
Comment (c-harding) [inline src/e2e/resources/simple-temple-expected/push-image.sh]: Ditto

File: src/main/resources/shell/deploy.sh.deployment.snippet [hunk 4]
```diff
@@ -1,5 +1,5 @@
 # Create private registry
-kubectl create -f kube/deploy
+kubectl create -f $BASEDIR/kube/deploy
```
Comment (c-harding) [inline src/main/resources/shell/deploy.sh.deployment.snippet]: Ditto

File: src/main/resources/shell/deploy.sh.deployment.snippet [hunk 5]
```diff
@@ -13,7 +13,10 @@ do
 done
 
 kubectl port-forward --namespace kube-system $REGISTRY_NAME 5000:5000 2>&1 1>/dev/null &
+PORT_FORWARD_PID=$!
 sleep 5
 
 # Push images to private repo
-sh push-image.sh
+sh $BASEDIR/push-image.sh
```
Comment (c-harding) [inline src/main/resources/shell/deploy.sh.deployment.snippet]: Ditto

**Review by c-harding — CHANGES_REQUESTED** [review]
Nice but needs more robustness. Not everyone has alphanumeric file names

**Review by c-harding — APPROVED** [review]

**Pull request merged**